### PR TITLE
Updates to Service Invocation and State Management

### PR DESCRIPTION
**Pull request by msfussell**
This PR is to address this issue https://github.com/dapr/docs/issues/362

### CODE REVIEW

**Comment by msfussell** [comment]
@yaron2 - Please review this PR and importantly provide detailed words for the Retries section. This

### Retries
Service invocation performs automatic retries with backoff time periods in the event of call failures and transient errors. 

People really want to know how retries work, so add plenty of detail and anything that they can configure (which I presume is nothing)

File: concepts/service-invocation/README.md [hunk 1]
```diff
@@ -1,32 +1,99 @@
...
-1. Service A makes a http/gRPC call meant for Service B.  The call goes to the local Dapr sidecar.  
-2. Dapr discovers Service B's location and forwards the message to Service B's Dapr sidecar
-3. Service B's Dapr sidecar forwards the request to Service B.  Service B performs its corresponding business logic.
-4. Service B sends a response for Service A.  The response goes to Service B's sidecar.
-5. Dapr forwards the response to Service A's Dapr sidecar.
-6. Service A receives the response.
+1. Service A makes an http/gRPC call meant for Service B.  The call goes to the local Dapr sidecar.  
+2. Dapr discovers Service B's location using the [name resolution component](https://github.com/dapr/components-contrib/tree/master/nameresolution) installed for the given hosting platform.
+3. Dapr forwards the message to Service B's Dapr sidecar 
+    * Note: All calls between Dapr sidecars go over gRPC for performance. Only calls from services to Dapr sidecars are either HTTP or gRPC
```
Comment (mukundansundar) [inline concepts/service-invocation/README.md:28]:
```suggestion
    * Note: All calls between Dapr sidecars go over gRPC for performance. Only calls between services and Dapr sidecars are either HTTP or gRPC
```

File: concepts/service-invocation/README.md [hunk 2]
```diff
@@ -1,32 +1,99 @@
...
+4. Service B's Dapr sidecar forwards the request to the specified endpoint (or method) on Service B.  Service B the runs its business logic code.
```
Comment (mukundansundar) [inline concepts/service-invocation/README.md:29]:
```suggestion
4. Service B's Dapr sidecar forwards the request to the specified endpoint (or method) on Service B.  Service B then runs its business logic code.
```

File: concepts/service-invocation/README.md [hunk 3]
```diff
@@ -1,32 +1,99 @@
...
+
+### Service-to-service security
+All calls between Dapr applications can be made secure with mutual (mTLS) authentication on hosted platforms, including automatic certificate rollover, via the Dapr Sentry service. The diagram below shows this for self hosted applications.
+
+For more information read the [service-to-service security](../security#mtls-self-hosted) article.
+
+![Self Hosted service to service security](../../images/security-mTLS-sentry-selfhosted.png)
+
+### Service access security
+Applications can control which other applications are allowed to call them and what they are authorized to do via access policies. This enables you to restrict sensitive applications, that say have personnel information, from being accessed by unauthorized applications and, combined with service-to-service secure communication, provides for soft multi-tenancy deployments.
```
Comment (mukundansundar) [inline concepts/service-invocation/README.md:83]:
```suggestion
Applications can control which other applications are allowed to call them and what they are authorized to do via access policies. This enables you to restrict sensitive applications, that say have personnel information, from being accessed by unauthorized applications, and combined with service-to-service secure communication, provides for soft multi-tenancy deployments.
```

File: howto/invoke-and-discover-services/README.md [hunk 4]
```diff
@@ -104,15 +96,21 @@ When running on [namespace supported platforms](../../reference/api/service_invo
 myApp.production
 ```
 
-See the [Cross namesapce API spec](../../reference/api/service_invocation_api.md#cross-namespace-invocation) for more information on namespaces.
+For example to invoke the example python service with a namespace would be;
```
Comment (mukundansundar) [inline howto/invoke-and-discover-services/README.md:99]:
```suggestion
For example, invoking the example python service with a namespace would be;
```

**Comment by yaron2** [comment]
@msfussell I updated the retries section.

**Comment by msfussell** [comment]
@yaron2 looks good. Can you approve the PR?

**Review by yaron2 — APPROVED** [review]
lgtm

**Pull request merged**